### PR TITLE
Fix applause sound stopping after switching scores

### LINKED ISSUE #12823: `applause` skin sound stops playing after switching to another score in beatmap leaderboard
**Describe the bug:** 
After finishing a beatmap, viewing another score stops the playback of `applause` skin sound.

This starts happening on `2021.514.0` and it is reproducible with default osu!lazer skin.

**Screenshots or videos showing encountered issue:** 

https://user-images.githubusercontent.com/52914632/118357072-67754480-b5a2-11eb-9606-268115c6e6ff.mp4

**osu!lazer version:** 
`2021.515.0`

**Logs:**
[performance.log](https://github.com/ppy/osu/files/6485689/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/6485691/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/6485693/performance-draw.log)
[performance-update.log](https://github.com/ppy/osu/files/6485695/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/6485697/runtime.log)
[updater.log](https://github.com/ppy/osu/files/6485698/updater.log)
[database.log](https://github.com/ppy/osu/files/6485699/database.log)
[network.log](https://github.com/ppy/osu/files/6485700/network.log)

### PR DESCRIPTION
**Pull request by happye921**
Fixes #12823

Moved the applause sound object from the AccuracyCircle to the ResultScreen. Timed the delay so it matches the old one.

### CODE REVIEW

**Comment by bdach** [comment]
@vininew921 it is not required to close pull requests after checks fail. you can reopen and push to the same branch you PRed this from (on your fork) to fix them.

File: osu.Game/Screens/Ranking/ResultsScreen.cs [hunk 1]
```diff
@@ -183,6 +195,11 @@ protected override void LoadComplete()
                 api.Queue(req);
 
             statisticsPanel.State.BindValueChanged(onStatisticsStateChanged, true);
+
+            using (BeginDelayedSequence(AccuracyCircle.ACCURACY_TRANSFORM_DELAY + AccuracyCircle.TEXT_APPEAR_DELAY, true))
+            {
+                this.Delay(-1000).Schedule(() => applauseSound?.Play());
```
Comment (peppy) [inline osu.Game/Screens/Ranking/ResultsScreen.cs:201]: why has the number changed?

Comment (bdach) [reply osu.Game/Screens/Ranking/ResultsScreen.cs:201]: I've prodded at this for a little bit, and while this number *does* need changing, I believe it should have changed by less, namely to

```suggestion
                this.Delay(-1140).Schedule(() => applauseSound?.Play());
```

Reasoning for this particular value is this:

https://github.com/ppy/osu/blob/41d8bc291ce81e496c660b8362567b2d9b1716a6/osu.Game/Screens/Ranking/ScorePanel.cs#L236-L237

The accuracy meter, along with the rest of the content, only starts showing after the panels have expanded. `resize_duration` is 200ms, `top_layer_expand_delay` is 100ms. Hence

```
-1440 + 200 + 100 = -1140
```

Can be roughly compared using `TestSceneResultsScreen` after setting playback speed low (~15% is best). The audio should start playing very soon after the accuracy meter appears and starts filling. Seemed more correct, can make videos of this if necessary.

**Review by bdach — DISMISSED** [review]

**Comment by peppy** [comment]
Which is fine, but all numbers should be drawn from constants, and match with `master`.

**Comment by bdach** [comment]
I'm not sure I follow. The adjusted value I suggested in my review comment *is* the same value, temporally. The number changes because the delay starts in a different component, with a different lifetime start, so delays are naturally affected. And the original constant is one that comes out of thin air.

I can make videos to show that it's the same, but at the end of the day, from the context I have the best that can be done is `-1440 + resize_duration + top_layer_expand_delay`. I'm not sure how to derive `-1440` originally.

Am I missing some context again?

**Comment by peppy** [comment]
I want to be able to review this by only reading the code. each portion of the duration which needs to be offset should be from a shared constant so it's clear why the value is what it is. And the original `1440` should be visible.

**Comment by bdach** [comment]
Sure, okay. So I guess I factor out `-1440` to a public constant on `AccuracyCircle`, expose the other two ones I need from `ScorePanel`, and combine those in `ResultsScreen`? Is that right?

Although even then it wouldn't be immediately visible why `resize_duration + top_layer_expand_delay` is a thing from the diff. I'd probably need to expose a sum of those and use that in both places where it needs to go (one at existing site in `ScorePanel`, one in `ResultsScreen`).

**Comment by peppy** [comment]
If this is decided as the correct direction for fixing this issue, yep, that sounds fine.

I'd hope we could come up with something more contained, but I haven't looked into it yet.

**Comment by bdach** [comment]
I can't say I'm seeing an alternative as of right now, at least one that isn't a variant on `AlwaysPresent`. And it does make more sense to have an applause sound effect at screen-level to me than down in the accuracy circle.

I guess I'll just go make changes as proposed above and if you can figure out something better, it can be opened as a separate PR superseding this one.

File: osu.Game/Screens/Ranking/ResultsScreen.cs (hunk 1 above)
Comment (vininew921) [inline osu.Game/Screens/Ranking/ResultsScreen.cs:201]: So as suggested I exposed the two variables on `ScorePanel` and made a new public constant on the `AccuracyCircle` called `APPLAUSE_DELAY`, so it's a bit clearer where the delay is coming from, as per your suggestions.

```suggestion
                this.Delay(ScorePanel.RESIZE_DURATION + ScorePanel.TOP_LAYER_EXPAND_DELAY - AccuracyCircle.APPLAUSE_DELAY).Schedule(() => applauseSound?.Play());
```

**Comment by vininew921** [comment]
Not sure why the CI check failed, I ran the same tests on my end and everything worked fine. I'm not that familiar with CI, so I don't know if it was just a fluke or a real error.

**Comment by peppy** [comment]
I've added back test coverage in `TestSceneResultsScreen`. Let's go with this for now.

**Review by peppy — APPROVED** [review]

**Pull request merged**